src/fs.zig

```diff
@@ -83,14 +83,14 @@ pub const FileSystem = struct {
         });
     }
 
-    pub var max_fd: FileDescriptorType = 0;
+    pub var max_fd: FileDescriptorType = .zero;
 
     pub inline fn setMaxFd(fd: anytype) void {
         if (!FeatureFlags.store_file_descriptors) {
             return;
         }
 
-        max_fd = @max(fd, max_fd);
+        max_fd = @enumFromInt(@max(fd, max_fd.int()));
```

Comment:
treating an FD like a number (max fd) case will (and has not ever) work on windows, because on windows is it not a number; there is not a great notion of "the highest fd".

thanks for uncovering this. i dont think it needs to be addressed yet because we dont set `store_file_descriptors` on windows.